src/mol-plugin/ui/controls/Canvas/CanvasComponent.tsx

```diff
@@ -0,0 +1,576 @@
+        //     x1: this.padding/2,
+        //     x2: this.width+this.padding/2,  
+        //     y1: this.padding/2,
+        //     y2: this.padding/2
+        // };
+        // this.rightLine = {
+        //     x1: this.width+this.padding/2,
+        //     x2: this.width+this.padding/2,
+        //     y1: this.padding/2,
+        //     y2: this.height+this.padding/2
+        // };
+        // this.bottomLine = {
+        //     x1: this.padding/2,
+        //     x2: this.width+this.padding/2,
+        //     y1: this.height+this.padding/2,
+        //     y2: this.height+this.padding/2
+        // };
+        // this.leftLine = {
+        //     x1: this.padding/2,
+        //     x2: this.padding/2, 
+        //     y1: this.padding/2,
+        //     y2: this.height+this.padding/2
+        // };
+        // this.normalizedLabel = this.normalizeGraphLabel(
+        //     this.label, 
+        //     this.padding/2, 
+        //     this.height+this.padding/2, 
+        //     this.width+this.padding/2, 
+        //     this.height+this.padding
+        // );
+        // this.normalizedInfoPoint = this.normalizeGraphLabel(
+        //     this.infoPoint, 
+        //     this.padding/2, 
+        //     this.height+this.padding/2, 
+        //     this.width+this.padding/2, 
+        //     this.height+this.padding
+        // );
+        // this.normalizedInfoLabel = this.normalizeGraphLabel(
+        //     this.infoLabel, 
+        //     this.padding/2, 
+        //     this.height+this.padding/2, 
+        //     this.width+this.padding/2, 
+        //     this.height+this.padding
+        // );
+        // this.normalizedRect = this.normalizeGraphLabel(
+        //     this.rainbowRect,
+        //     this.padding/2,
+        //     0,
+        //     this.width+this.padding/2,
+        //     this.padding/2
+        // );
+        
+        for (const point of this.props.data){
+            this.state.points.push(point);
+        }
+        
+        this.state.points.sort((a, b) => { 
+            if(a[0] === b[0]){
+                if(a[0] === 0){
+                    return a[1]-b[1];
+                }
+                if(a[1] === 1){
+                    return b[1]-a[1];
+                }
+                return a[1]-b[1];
+            }
+            return a[0] - b[0];
+        });
+
+        this.handleDrag = this.handleDrag.bind(this);
+        this.handleClickCanvas = this.handleClickCanvas.bind(this);
+        this.handleDoubleClick = this.handleDoubleClick.bind(this);
+        this.refCallBack = this.refCallBack.bind(this);
+        this.handlePointUpdate = this.handlePointUpdate.bind(this);
+        this.change = this.change.bind(this);
+    }
+
+    public render() {
+        const points = this.renderPoints();
+        const ghostPoint = this.state.copyPoint;
+        const selectedPoint = this.state.selectedPoint;
+        return ([
+            <div key="canvas" id="canvas">                
```

Comment:
please don't use hardcoded id's as it breaks things if you have multiple instances of the component